source/Coop/Mod/DebugUtil/Replay.cs

```diff
@@ -0,0 +1,413 @@
+﻿using Coop.Mod.Persistence;
+using Coop.Mod.Persistence.Party;
+using NLog;
+using RailgunNet.System.Encoding;
+using RailgunNet.System.Types;
+using RailgunNet.Util;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.Library;
+using Path = System.IO.Path;
+
+namespace Coop.Mod.DebugUtil
+{
+    /// <summary>
+    /// How it works:
+    /// <list type="bullet">
+    /// <item>load savefile in game</item>
+    /// <item>start server</item>
+    /// <item>open 'RGL Command line console'</item>
+    /// <item>(1) type 'coop.record 1'</item>
+    /// <item>do some movements on campaign map</item>
+    /// <item>type 'coop.stop'</item>
+    /// <item>stop server</item>
+    /// <item>load the same savefile in game</item>
+    /// <item>start server</item>
+    /// <item>open rgl-debug-console</item>
+    /// <item>(2) type 'coop.play 1'</item>
+    /// <item>setup normal speed (key '2')</item>
+    /// <item>main hero party will be moving exactly same as past</item>
+    /// <item>(3) wait while in log shown '[REPLAY] Verifying has finished ...'</item>
+    /// <item>view file <c>\mb2\bin\Win64_Shipping_Client\logs\1-verified.html</c> in browser or Excel</item>
+    /// </list>
+    /// Column names used in report:
+    /// <list type="bullet">
+    /// <item><term>timeDiff(ms)</term><description> it is the in-game time interval in ms between
+    /// related saved and replayed movement events</description></item>
+    /// <item><term>skipped</term><description> it is the number of saved movement which did not find
+    /// a related movement while playback</description></item>
+    /// <item><term>newEvents</term><description> it is the number of new movement happend while playback
+    /// which did not find a related in saved movements</description></item>
+    /// <item><term>cols</term><description> it is the statistical info about distribution of time intervals
+    /// calculated as <c>count(log2(time_diff), column)</c></description></item>
+    /// </list>
+    /// Constant shift in result data no matter.
+    /// It is important that the spread of values is minimal.
+    /// See <seealso>https://github.com/Bannerlord-Coop-Team/BannerlordCoop/pull/33</seealso>
+    /// </summary>
+    public static class Replay
```

Comment:
A short summary is sufficient. This should be added to the wiki. You can reference it in remarks